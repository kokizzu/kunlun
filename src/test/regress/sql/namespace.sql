--
-- Regression tests for schemas (namespaces)
--

--DDL_STATEMENT_BEGIN--
CREATE SCHEMA test_ns_schema_1;
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
CREATE TABLE test_ns_schema_1.abc (
              a serial,
              b int UNIQUE
);
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
CREATE UNIQUE INDEX abc_a_idx ON test_ns_schema_1.abc (a);
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
CREATE VIEW test_ns_schema_1.abc_view AS SELECT a+1 AS a, b+1 AS b FROM test_ns_schema_1.abc;
--DDL_STATEMENT_END--

-- verify that the objects were created
SELECT COUNT(*) FROM pg_class WHERE relnamespace =
    (SELECT oid FROM pg_namespace WHERE nspname = 'test_ns_schema_1');

INSERT INTO test_ns_schema_1.abc DEFAULT VALUES;
INSERT INTO test_ns_schema_1.abc DEFAULT VALUES;
INSERT INTO test_ns_schema_1.abc DEFAULT VALUES;

SELECT * FROM test_ns_schema_1.abc;
SELECT * FROM test_ns_schema_1.abc_view;

--DDL_STATEMENT_BEGIN--
DROP TABLE test_ns_schema_1.abc cascade;
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
DROP SCHEMA test_ns_schema_1 CASCADE;
--DDL_STATEMENT_END--