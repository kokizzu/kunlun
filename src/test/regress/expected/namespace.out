--
-- Regression tests for schemas (namespaces)
--
CREATE SCHEMA test_ns_schema_1;
CREATE SCHEMA
CREATE TABLE test_ns_schema_1.abc (
              a serial,
              b int UNIQUE
);
CREATE TABLE
CREATE UNIQUE INDEX abc_a_idx ON test_ns_schema_1.abc (a);
CREATE INDEX
CREATE VIEW test_ns_schema_1.abc_view AS SELECT a+1 AS a, b+1 AS b FROM test_ns_schema_1.abc;
CREATE VIEW
-- verify that the objects were created
SELECT COUNT(*) FROM pg_class WHERE relnamespace =
    (SELECT oid FROM pg_namespace WHERE nspname = 'test_ns_schema_1');
 count 
-------
     5
(1 row)

INSERT INTO test_ns_schema_1.abc DEFAULT VALUES;
INSERT 0 1
INSERT INTO test_ns_schema_1.abc DEFAULT VALUES;
INSERT 0 1
INSERT INTO test_ns_schema_1.abc DEFAULT VALUES;
INSERT 0 1
SELECT * FROM test_ns_schema_1.abc;
 a | b 
---+---
 1 |  
 2 |  
 3 |  
(3 rows)

SELECT * FROM test_ns_schema_1.abc_view;
 a | b 
---+---
 2 |  
 3 |  
 4 |  
(3 rows)

DROP TABLE test_ns_schema_1.abc cascade;
psql:sql/namespace.sql:26: NOTICE:  drop cascades to view test_ns_schema_1.abc_view
DROP TABLE
DROP SCHEMA test_ns_schema_1 CASCADE;
DROP SCHEMA
